sdk/CONTRIBUTING.md

```diff
@@ -1,12 +1,69 @@
 ## Contributing to the `kfp` SDK
-KFP v2 for SDK is under development on the `master` branch.
 
-To contribute to v1, please go to [sdk/release-1.8 branch](https://github.com/kubeflow/pipelines/tree/sdk/release-1.8) to submit your change.
+For developing KFP v2 for SDK use the `master` branch.
+
+For developing KFP v1 for SDK use the [sdk/release-1.8 branch](https://github.com/kubeflow/pipelines/tree/sdk/release-1.8) to submit your changes.
 
 For general contribution guidelines including pull request conventions, see [pipelines/CONTRIBUTING.md](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md).
 
-### Requirements
-All development requirement versions are pinned in [requirements-dev.txt](https://github.com/kubeflow/pipelines/blob/master/sdk/python/requirements-dev.txt).
+### Pre-requisites 
```

Comment:
We have the pre-requisites as a bash script: https://github.com/kubeflow/pipelines/blob/master/test/presubmit-test-kfp-kubernetes-library.sh
Today the script has two main responsibilities:
- Setup the local environment 
- Execute the test

Adding the text in this file is duplicating the effort while doing the maintenance. That script is used in the GHA to ensure that everything goes well while submitting a change upstream.

I see areas of improvement:
- Create a new script called `presubmit-kfp-kubernetes-library.sh`
- Change `presubmit-test-kfp-kubernetes-library.sh` to do a call to `presubmit-kfp-kubernetes-library.sh`
- Make `presubmit-kfp-kubernetes-library.sh` agnostic to support Fedora and Ubuntu
- Add a reference in this file to call `presubmit-kfp-kubernetes-library.sh` before starting working or while doing a test locally

Reply:
The content of `presubmit-kfp-kubernetes-library.sh` is:
```
source_root=$(pwd)

pip install --upgrade pip
pip install wheel

pip install sdk/python
apt-get update && apt-get install -y protobuf-compiler
pushd api
make clean python
popd
python3 -m pip install api/v2alpha1/python
pip install api/v2alpha1/python

# generate Python proto code from source
pushd "$source_root/kubernetes_platform"
make clean python
popd

# rust needed for transitive deps in dev extras on Python:3.12
apt-get install rustc -y
pip install -e "$source_root/kubernetes_platform/python[dev]"
```

As I mentioned above, you should make it agnostic to support Ubuntu or Fedora

Reply:
I think that's a good suggestion, but out of scope of this pr.

Reply:
I don't see it as out of the scope of the PR. We are introducing duplicated information

Reply:
changing ci, and pre-commit tooling is definitely a task in it's own right and out of scope of this is simple pr to just add more documentation